src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/AmbariClusterConnector.java

```diff
@@ -132,12 +142,18 @@ public void buildAmbariCluster(Stack stack) {
             Map<String, List<String>> hostGroupMappings = recommend(stack, ambariClient);
             saveHostMetadata(cluster, hostGroupMappings);
             ambariClient.createCluster(cluster.getName(), blueprint.getBlueprintName(), hostGroupMappings);
-            waitForClusterInstall(stack, ambariClient);
-            runSmokeTest(stack, ambariClient);
-            if (cluster.getRecipe() != null) {
-                recipeEngine.executePostInstall(stack);
+            PollingResult pollingResult = waitForClusterInstall(stack, ambariClient);
+            if (pollingResult.equals(PollingResult.SUCCESS)) {
```

Comment:
I think we should use the same form to check if the result was success or exit. Create a method in the PollingResult isSuccess() and isExited() and use that method in each if condition